e2e/quota/quota_enable.go

```diff
@@ -0,0 +1,119 @@
+package e2e
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/restclient"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	volname = "testvol"
+)
+
+var (
+	gds    []*gdProcess
+	client *restclient.Client
+	tmpDir string
+)
+
+// TestVolume creates a volume and starts it, runs further tests on it and
+// finally deletes the volume
+func TestVolume(t *testing.T) {
+	var err error
+
+	r := require.New(t)
+
+	gds, err := setupCluster("./config/1.yaml", "./config/2.yaml")
+	r.Nil(err)
+	defer teardownCluster(gds)
+
+	client := initRestclient(gds[0].ClientAddress)
+
+	tmpDir, err = ioutil.TempDir("", t.Name())
+	r.Nil(err)
+	t.Logf("Using temp dir: %s", tmpDir)
+	//defer os.RemoveAll(tmpDir)
+
+	// Create the volume
+	t.Run("Create", testVolumeCreate)
+
+	// Run tests that depend on this volume
+	t.Run("Start", testVolumeStart)
+	t.Run("Status", testVolumeStatus)
+	t.Run("Quota-enable", testQuotaEnable)
+	t.Run("Stop", testVolumeStop)
+
+	// delete volume
+	t.Run("Delete", testVolumeDelete)
+
+}
+
+func testVolumeCreate(t *testing.T) {
+	r := require.New(t)
+
+	var brickPaths []string
+	for i := 1; i <= 4; i++ {
+		brickPath, err := ioutil.TempDir(tmpDir, "brick")
+		r.Nil(err)
+		brickPaths = append(brickPaths, brickPath)
+	}
+
+	// create 2x2 dist-rep volume
+	createReq := api.VolCreateReq{
+		Name: volname,
+		Subvols: []api.SubvolReq{
+			{
+				ReplicaCount: 2,
+				Type:         "replicate",
+				Bricks: []api.BrickReq{
+					{NodeID: gds[0].PeerID(), Path: brickPaths[0]},
+					{NodeID: gds[1].PeerID(), Path: brickPaths[1]},
+				},
+			},
+			{
+				Type:         "replicate",
+				ReplicaCount: 2,
+				Bricks: []api.BrickReq{
+					{NodeID: gds[0].PeerID(), Path: brickPaths[2]},
+					{NodeID: gds[1].PeerID(), Path: brickPaths[3]},
+				},
+			},
+		},
+		Force: true,
+	}
+	_, err := client.VolumeCreate(createReq)
+	r.Nil(err)
+}
+
+func testVolumeDelete(t *testing.T) {
```

Comment:
Each tests may be executed in parallel or in different order. Make this as single test as `testQuotaEnable` and have all this in that function

Reply:
done